keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1150 @@
+    //
+    // +optional
+    // +listType=atomic
+    Options []resource.Quantity
+}
+
+
+// CapacitySharingPolicyRange defines a valid range for consumable capacity values.
+//
+// - If the requested amount is less than Minimum, it is rounded up to the Minimum value.
+// - If the requested amount is between Minimum and Maximum, ChunkSize is set,
+//   and the amount does not align with the ChunkSize,
+//   it will be rounded up to the next value matching Minimum + (n * ChunkSize).
+// - If the requested or rounded amount exceeds Maximum (if set), the request does not satisfy the policy,
+//   and the device cannot be allocated.
+//
+// - If ChunkSize is not set, the requested amount is used as-is, provided it falls within the range.
+type CapacitySharingPolicyRange struct {
+    // Minimum specifies the minimum capacity allowed for a consumption request.
+    // This also acts as the default if no value is specified.
+    //
+    // +required
+    Minimum resource.Quantity
+
+    // Maximum defines the upper limit for capacity that can be requested.
+    //
+    // +optional
+    Maximum *resource.Quantity
+
+    // ChunkSize defines the step size between valid capacity amounts within the range.
+    // If set, requested amounts are rounded up to the nearest multiple of ChunkSize from the Minimum.
+    //
+    // +optional
+    ChunkSize *resource.Quantity
+}
+```
+
+#### ResourceClaimSpec's DeviceRequest
+
+```go
+
+type DeviceRequest struct {
+...
+   // CapacityRequests define resource requirements against each capacity.
+   //
+   // +optional
+   // +featureGate=DRAConsumableCapacity
+   CapacityRequests *CapacityRequirements
+}
+
+
+// CapacityRequirements defines the capacity requirements for a specific device request.
+type CapacityRequirements struct {
+    // Minimum defines the minimum amount of each device capacity required for the request.
+    //
+    // If the capacity has a sharing policy, this value is rounded up to the nearest valid amount
+    // according to that policy. The rounded value is used during scheduling to determine how much capacity to consume.
+    //
+    // If the quantity does not have a sharing policy, this value is used as an additional filtering
+    // condition against the available capacity on the device.
+    // This is semantically equivalent to a CEL selector with
+    // `device.quantity[<domain>].<name>.compareTo(<minimum quantity>) >= 0`.
+    //
+    // +optional
+   Minimum map[QualifiedName]resource.Quantity
+}
+
+// DeviceConstraint must have exactly one field set besides Requests.
+type DeviceConstraint struct {
+	...
+   // DistinctAttribute requires that all devices in question have this
+   // attribute and that its type and value are unique across those devices.
+   //
+   // This acts as the inverse of MatchAttribute.
+   //
+   // This constraint is used to avoid allocating multiple requests to the same shareable device
+   // by ensuring attribute-level differentiation.
```

Comment:
Is there an example scenario where allocating multiple requests to the same device while satisfying all the requested capacities would be problematic, but it allocating requests from different resourceclaims to the same device would be ok? I'm trying to understand the use case for this (what breaks without this)

Reply:
My understanding of the primary use case here is spread for availability - "give me two network interfaces but make sure they are allocated out of two separate physical NICs".

Reply:
ok... I can see how that's useful ... is that related to sharing?

Reply:
Not super directly as far as I can tell...thus my comment in [here](https://github.com/kubernetes/enhancements/pull/5104#discussion_r2100678577).

Reply:
I guess this KEP makes it newly possible for the same device to be allocated to multiple requests in the same ResourceClaim, and without this, the claim cannot prevent that, so it *is* vaguely related